statsmodels/regression/linear_model.py

```diff
@@ -330,6 +332,37 @@ def fit(self, method="pinv", cov_type='nonrobust', cov_kwds=None,
             # used in ANOVA
             self.effects = effects = np.dot(Q.T, self.wendog)
             beta = np.linalg.solve(R, effects)
+        elif method == "qr-pivot":
+            Q, R, P = linalg.qr(self.wexog, mode='economic', pivoting=True)
+            self.exog_Q, self.exog_R, self.exog_P = Q, R, P
+
+            # Cache singular values from R.
+            self.wexog_singular_values = np.linalg.svd(R, 0, 0)
+            tol = kwargs.get('tol', 1e-7)
+
+            singular_preds = np.abs(np.diagonal(R)) < tol
+            n_singular = np.sum(singular_preds)
+            n_preds = len(singular_preds)
+            self.rank = n_preds - n_singular
+            sortP = np.argsort(P)
+
+            if n_singular > 0:
+                rng = slice(self.rank)
+                Qq = Q[:, rng]
+                Rr = R[rng, :][:, rng]
+                effects = np.dot(Qq.T, self.wendog)
+                beta = linalg.solve_triangular(Rr, effects)
+                beta = np.concatenate((beta, np.repeat(0.0, n_singular)))[sortP]
+                self.effects = np.concatenate((effects, np.repeat(np.NaN, n_singular)))
```

Comment:
this will need unit tests for anova_lm
I don't know if this should be nan or zero or something else